src/components/BulkEnrollmentModal/index.jsx

```diff
@@ -0,0 +1,219 @@
+      errorMessageRef.focus();
+    }
+  }
+
+  handleModalSubmit(formData) {
+    const {
+      enterpriseUuid,
+      sendBulkEnrollment,
+      selectedCourseRunKeys,
+    } = this.props;
+
+    const bulkEnrollmentPromises = selectedCourseRunKeys.map((selectedCourseRunKey) => {
+      const options = {
+        course_run_key: selectedCourseRunKey,
+        email: formData['bulk-enrollment-email-addresses'],
+        email_csv: formData['bulk-enrollment-email-csv'],
+      };
+      return sendBulkEnrollment(enterpriseUuid, options);
+    });
+
+    return Promise.all(bulkEnrollmentPromises)
+      .then((response) => {
+        this.props.onSuccess(response);
+      })
+      .catch((error) => {
+        throw new SubmissionError({
+          _error: [error.message],
+        });
+      });
+  }
+
+  resetModalWrapperState() {
+    this.setState({ open: false });
+  }
+
+  renderErrorMessage() {
+    const { error } = this.props;
+
+    return (
+      <div
+        ref={this.errorMessageRef}
+        tabIndex="-1"
+      >
+        <Alert
+          variant="danger"
+          dismissible
+        >
+          <Alert.Heading>Unable to assign codes</Alert.Heading>
+          <p className="alert-body">{error}</p>
+        </Alert>
+      </div>
+    );
+  }
+
+  renderFailedLearnersMessage(failedLearners) {
+    return (
+      <div
+        ref={this.errorMessageRef}
+        tabIndex="-1"
+      >
+        <Alert
+          variant="danger"
+          dismissible
+        >
+          <Alert.Heading>
+            Before you can invite learners to a course they must have a subscription license
+            </Alert.Heading>
+          <p className="alert-body">
+            The following emails addresses are not yet associated with a subscription.
+            Please take note of the following learners and invite them from the subscription management page.
+          </p>
+          {failedLearners.map((failedLearner) => <p className="alert-email mb-0">{failedLearner}</p>)}
+        </Alert>
+      </div>
+    );
+  }
+
+  renderBody() {
+    const {
+      failedLearners,
+      selectedCourseRunKeys,
+      submitFailed,
+    } = this.props;
+
+    return (
+      <>
+        {submitFailed && this.renderErrorMessage()}
+        {failedLearners.length > 0 && this.renderFailedLearnersMessage(failedLearners)}
+        <p className="courses-selected">Courses selected: {selectedCourseRunKeys.length}</p>
+        <form onSubmit={e => e.preventDefault()}>
+          <div className="mt-4">
+            <h3>Add Learners</h3>
+            <div>
+              <Field
+                id="bulk-enrollment-email-addresses"
+                name="bulk-enrollment-email-addresses"
+                component={TextAreaAutoSize}
+                label="Email addresses"
+                type="text"
+              />
+              <p>To add more than one user, enter one email address per line</p>
```

Comment:
Check with Ange, but I think we probably want periods at the end of this microcopy.